### DISCUSSION TITLE
How to update list after adding new item?

### DISCUSSION THREAD

**Discussion by villesau** [urql-graphql/urql#2214] [Q&A] [answered]
urql has pretty effortless updates of individual items when mutating them. But what is the best practice when adding new items and you have queries representing lists of them? So when I have list of items, and a I add new ones, I'd need to refresh the list, or pull the mutated item and add that to the list. Is there a way for urql to help with that like with mutating of existing items?

**Comment by JoviDeCroock** [comment, marked as answer]
Assuming you are talking about graphcache:

- [You can link items or update the query](https://formidable.com/open-source/urql/docs/graphcache/cache-updates/#updating-lists-or-links)
- [You can invalidate the list](https://formidable.com/open-source/urql/docs/graphcache/cache-updates/#updating-many-unknown-links)

(Paginated) Lists in GraphQL are always a bit hard so we're still looking for better paradigms in case some come to mind :)

**Reply by villesau** [reply]
Thank you! How would that work with graphql-codegen? I see that it generates Document variables, but they are full queries while, to my understanding based on the todo-example, it should only have the field `todos`. So what graphql-codegen gives:

```
export const GetTodosDocument = gql`
    query GetTodos {
  todos {
    id
  }
}
    `;
```

while it should only give 

```
export const GetTodosDocument = gql`
{
  todos {
    id
  }
}
    `;
```

Might also be that I'm just missing something here.

**Reply by JoviDeCroock** [reply]
Both are the same thing 😅 you can omit both the query keyword as well as the operationName, the only thing that matters here is the rootField to field connection (todos)

**Reply by villesau** [reply]
For some reason I'm not able to pull anything out of cache. I always get null as a result :/ For both: `cache.resolve('Query', 'todos');` and

```
        const TodoList = gql`
          {
            todos {
              id
            }
          }
        `;

        cache.updateQuery({ query: TodoList }, data => {
          data.todos.push(result.createTodo);
          return data;
        });
```

**Reply by JoviDeCroock** [reply]
It's hard to just pinpoint the issue but I can ask some general questions to try, what kind off field are you targetting i.e. if your list has arguments then it won't be as easy as just saying `todos` as we probably want `todos(limit: x)`, you can see what's under the root-query field by running `cache.inspectFields('Query')`.

A second posibility is that you just don't have any `todos` in your cache butonly a `todo`? 😅 don't really know the case you're trying to show so just taking some guesses

Can always refer to https://github.com/FormidableLabs/urql/blob/main/examples/with-graphcache-updates/src/client.js

**Reply by villesau** [reply]
Omg, doh. Yeah, I did had an argument there :) Just blindly followed the todo case so this didn't cross my mind. Thanks for the help! 😅

So it would've been something like this in my case:

```
            const res = cache.resolve("Query", "todos", {
              limit: x,
            });
```

**Reply by JoviDeCroock** [reply]
Hehe, awesome 🙌
